src/Components/App/App.js

```diff
@@ -18,36 +19,56 @@ const App = () => {
       } catch (error) {
         console.log(error);
       }
-    }
+    };
     fetchAllPokemon();
   }, []);
 
   const get4RdmPokemon = () => {
-    const pokemons = [{}, {}, {}, {}]
-    let randomIndexes = []
-    while(randomIndexes.length <= 4) {
+    const pokemons = [{}, {}, {}, {}];
+    let randomIndexes = [];
+    while (randomIndexes.length <= 4) {
       let randNum = Math.floor(Math.random() * allPokemon.length);
       if (randomIndexes.indexOf(randNum) === -1) randomIndexes.push(randNum);
     }
-    return pokemons.map((emptySlot,i) => {
-      const randomIndex = randomIndexes[i]
+    return pokemons.map((emptySlot, i) => {
+      const randomIndex = randomIndexes[i];
       return allPokemon[randomIndex];
-    })
+    });
+  };
+
+  const fetchSinglePokemon = async (url) => {
+    try {
+      const singlePokemon = await getSinglePokemon(url);
+      setSinglePokemon(singlePokemon);
+    } catch (error) {
+      console.log(error);
```

Comment:
Don't forget to implement some more powerful error handling -- don't expect the user to open their console if something goes wrong.